README.md

```diff
@@ -4,41 +4,185 @@ Please edit this README.md file with your answers to the worksheet questions.
+|A | B | (A AND B) | NOT (A AND B)|
+|:-|:-:|:---------:| ------------:|     
+|F | F |    F      |     T        |
+|F | T |    F      |     T        |
+|T | F |    F      |     T        |
+|T | T |    T      |     F        |
+
+|NOT A | NOT B | NOT A OR NOT B|
+|:-----|:-----:| -------------:|      
+|  T   |   T   |       T       |
+|  T   |   F   |       T       |
+|  F   |   T   |       T       |
+|  F   |   F   |       F       |
+
+### c 
+|A | B | C | (A AND B) | (A AND C) | (A AND B) OR (A AND C)|
+|:-|:-:|:-:|:---------:|:---------:| ---------------------:|    
+|F | F | F |    F      |     F     |          F            |
+|F | F | T |    F      |     F     |          F            |
+|F | T | F |    F      |     F     |          F            |
+|F | T | T |    F      |     F     |          F            |
+|T | F | F |    F      |     F     |          F            |
+|T | F | T |    F      |     T     |          T            |
+|T | T | F |    T      |     F     |          T            |
+|T | T | T |    T      |     T     |          T            |
+
+|(B OR C) | A AND (B OR C)|
+|:--------|--------------:|    
+|   F     |      F        |
+|   T     |      F        |
+|   T     |      F        |
+|   T     |      F        |
+|   F     |      F        |
+|   T     |      T        |
+|   T     |      T        |
+|   T     |      T        |
+
+### d 
+|A | B | C | (A OR B) | (A OR C) | (A OR B) AND (A OR C)|
+|:-|:-:|:-:|:--------:|:--------:|---------------------:|       
+|F | F | F |    F     |     F    |           F          |
+|F | F | T |    F     |     T    |           F          |
+|F | T | F |    T     |     F    |           F          |
+|F | T | T |    T     |     T    |           T          |
+|T | F | F |    T     |     T    |           T          |
+|T | F | T |    T     |     T    |           T          |
+|T | T | F |    T     |     T    |           T          |
+|T | T | T |    T     |     T    |           T          |
+
+|(B AND C) | A OR (B AND C)|
+|:---------|--------------:|    
+|    F     |       F       |
+|    F     |       F       |
+|    F     |       F       | 
+|    T     |       T       |
+|    F     |       T       |
+|    F     |       T       |
+|    F     |       T       |
+|    T     |       T       |
 
 ## Question 4
 
-### a
-
-### b
-
-### c
+### a The progrmas in A are equivilent becasue NOT(A AND B) = NOT A AND NOT B. You can see the resemblance between that and the pair of programs. 
+
+### b The pair of programs are the same in B becasue (A AND B) OR (A AND C) = A AND (B OR C). Further up the page, you can see why (A AND B) OR (A AND C) would give out the same values as A AND (B OR C).
+
+### c The pair of programs will put out oposite answers becasue A AND B and NOT A OR NOT B will put out oposite answers. The programs would be the same as A AND B and NOT A OR NOT B if you replaced the x and y values with A and B. A AND B and NOT A OR NOT B can be shown in the truth table I have drawn up here: 
```

Comment:
Note that the if and else parts of the two programs are switched around -- so even though the boolean expressions are reversed, the behaviour of the program is the same.